We should always use MP config in CDI extension

FaultToleranceCdiExtension uses `Config.create()` which should not be done. This should use the current MP config (and convert it to SE config if desired)

To replicate - create an invalid `meta-config.yaml` on classpath, with for example the following content:
```yaml
sources:
  - type: "wrong"
```
Create a custom main class and configure MP config as follows:
```java
ConfigProviderResolver configProvider = ConfigProviderResolver.instance();

configProvider.registerConfig(configProvider.getBuilder()
                                                // system properties, env vars and META-INF/microprofile-config.properties
                                              .addDefaultSources()
                                              .build(),
                                      Thread.currentThread().getContextClassLoader());

io.helidon.microprofile.cdi.Main.main(args);
```

This should make sure the registered config is used by all components. It fails now with
```
Caused by: java.lang.IllegalArgumentException: Config source of type wrong is not supported. Supported types: [file, prefixed, classpath, system-properties, directory, url, environment-variables]
...
at io.helidon.microprofile.faulttolerance.FaultToleranceExtension.registerMetricsAndInitExecutors(FaultToleranceExtension.java:273)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:97)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	
```

--- a/integrations/micrometer/cdi/src/main/java/io/helidon/integrations/micrometer/cdi/MeterRegistryProducer.java
+++ b/integrations/micrometer/cdi/src/main/java/io/helidon/integrations/micrometer/cdi/MeterRegistryProducer.java
@@ -21,10 +21,12 @@ import javax.enterprise.inject.Produces;
 
 import io.helidon.common.LazyValue;
 import io.helidon.config.Config;
+import io.helidon.config.mp.MpConfig;
 import io.helidon.integrations.micrometer.MeterRegistryFactory;
 import io.helidon.integrations.micrometer.MicrometerSupport;
 
 import io.micrometer.core.instrument.MeterRegistry;
+import org.eclipse.microprofile.config.ConfigProvider;
 
 @ApplicationScoped
 class MeterRegistryProducer {
@@ -55,7 +57,7 @@ class MeterRegistryProducer {
     }
 
     private static MicrometerSupport createMicrometerSupport() {
-        Config micrometerConfig = Config.create().get(CONFIG_KEY);
+        Config micrometerConfig = MpConfig.toHelidonConfig(ConfigProvider.getConfig()).get(CONFIG_KEY);
         MeterRegistryFactory factory = MeterRegistryFactory.getInstance(
                 MeterRegistryFactory.builder()
                     .config(micrometerConfig));

--- a/integrations/micrometer/cdi/src/main/java/module-info.java
+++ b/integrations/micrometer/cdi/src/main/java/module-info.java
@@ -35,6 +35,7 @@ module io.helidon.integrations.micrometer.cdi {
     requires io.helidon.config;
     requires io.helidon.webserver.cors;
     requires io.helidon.integrations.micrometer;
+    requires io.helidon.config.mp;
 
     requires micrometer.core;
     requires micrometer.registry.prometheus;

--- a/integrations/micrometer/micrometer/src/main/java/io/helidon/integrations/micrometer/MeterRegistryFactory.java
+++ b/integrations/micrometer/micrometer/src/main/java/io/helidon/integrations/micrometer/MeterRegistryFactory.java
@@ -341,37 +341,37 @@ public final class MeterRegistryFactory {
          * @param registriesConfig {@code Config} object for the 1 or more {@code builtin-registries} entries
          */
         private void enrollBuiltInRegistries(Config registriesConfig) {
-
-            if (registriesConfig.type() != Config.Type.LIST) {
-                throw new IllegalArgumentException("Expected Micrometer config " + BUILTIN_REGISTRIES_CONFIG_KEY + " as a LIST "
-                        + "but found " + registriesConfig.type().name());
-            }
-
+            // in case config is created from MP config, node type is OBJECT instead of LIST
+            // as MP config flattens everything
             Map<BuiltInRegistryType, MicrometerBuiltInRegistrySupport> candidateBuiltInRegistryTypes =
                     new HashMap<>();
             List<String> unrecognizedTypes = new ArrayList<>();
 
-            for (Config registryConfig : registriesConfig.asNodeList().get()) {
-                String registryType = registryConfig.get("type").asString().get();
-                try {
-                    BuiltInRegistryType type =
-                            BuiltInRegistryType.valueByName(registryType);
+            registriesConfig.asList(Config.class)
+                    .ifPresent(confList -> {
+                        for (Config registryConfig : confList) {
+                            String registryType = registryConfig.get("type").asString().get();
+                            try {
+                                BuiltInRegistryType type =
+                                        BuiltInRegistryType.valueByName(registryType);
 
-                    MicrometerBuiltInRegistrySupport builtInRegistrySupport =
-                            MicrometerBuiltInRegistrySupport.create(type, registryConfig.asNode());
-                    if (builtInRegistrySupport != null) {
-                        candidateBuiltInRegistryTypes.put(type, builtInRegistrySupport);
-                    }
-                } catch (BuiltInRegistryType.UnrecognizedBuiltInRegistryTypeException e) {
-                    unrecognizedTypes.add(e.unrecognizedType());
-                    logRecords.add(new LogRecord(Level.WARNING,
-                            String.format("Ignoring unrecognized Micrometer built-in registry type %s", e.unrecognizedType())));
-                }
-            }
+                                MicrometerBuiltInRegistrySupport builtInRegistrySupport =
+                                        MicrometerBuiltInRegistrySupport.create(type, registryConfig.asNode());
+
+                                candidateBuiltInRegistryTypes.put(type, builtInRegistrySupport);
+                            } catch (BuiltInRegistryType.UnrecognizedBuiltInRegistryTypeException e) {
+                                unrecognizedTypes.add(e.unrecognizedType());
+                                logRecords.add(new LogRecord(Level.WARNING,
+                                                             String.format(
+                                                                     "Ignoring unrecognized Micrometer built-in registry type %s",
+                                                                     e.unrecognizedType())));
+                            }
+                        }
+                    });
 
             if (!unrecognizedTypes.isEmpty()) {
                 LOGGER.log(Level.WARNING, String.format("Ignoring unrecognized Micrometer built-in registries: %s",
-                        unrecognizedTypes.toString()));
+                                                        unrecognizedTypes));
             }
 
             // Do not change previous settings if we did not find any valid new built-in registries selected.
@@ -379,7 +379,7 @@ public final class MeterRegistryFactory {
                 builtInRegistriesRequested.clear();
                 builtInRegistriesRequested.putAll(candidateBuiltInRegistryTypes);
                 LOGGER.log(Level.FINE,
-                        () -> "Selecting built-in Micrometer registries " + candidateBuiltInRegistryTypes.toString());
+                        () -> "Selecting built-in Micrometer registries " + candidateBuiltInRegistryTypes);
             }
         }
     }

--- a/microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/FaultToleranceExtension.java
+++ b/microprofile/fault-tolerance/src/main/java/io/helidon/microprofile/faulttolerance/FaultToleranceExtension.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2018, 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2018, 2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -44,6 +44,7 @@ import javax.inject.Inject;
 
 import io.helidon.common.configurable.ScheduledThreadPoolSupplier;
 import io.helidon.common.configurable.ThreadPoolSupplier;
+import io.helidon.config.mp.MpConfig;
 import io.helidon.faulttolerance.FaultTolerance;
 
 import org.eclipse.microprofile.config.Config;
@@ -270,7 +271,7 @@ public class FaultToleranceExtension implements Extension {
         }
 
         // Initialize executors for MP FT - default size of 16
-        io.helidon.config.Config config = io.helidon.config.Config.create();
+        io.helidon.config.Config config = MpConfig.toHelidonConfig(ConfigProvider.getConfig());
         scheduledThreadPoolSupplier = ScheduledThreadPoolSupplier.builder()
                 .threadNamePrefix("ft-mp-schedule-")
                 .corePoolSize(16)

--- a/microprofile/fault-tolerance/src/main/java/module-info.java
+++ b/microprofile/fault-tolerance/src/main/java/module-info.java
@@ -29,6 +29,7 @@ module io.helidon.microprofile.faulttolerance {
     requires io.helidon.microprofile.config;
     requires io.helidon.microprofile.server;
     requires io.helidon.microprofile.metrics;
+    requires io.helidon.config.mp;
 
     requires jakarta.enterprise.cdi.api;
 

--- a/microprofile/grpc/server/src/main/java/io/helidon/microprofile/grpc/server/GrpcServerCdiExtension.java
+++ b/microprofile/grpc/server/src/main/java/io/helidon/microprofile/grpc/server/GrpcServerCdiExtension.java
@@ -39,6 +39,7 @@ import javax.enterprise.inject.spi.Extension;
 
 import io.helidon.common.serviceloader.HelidonServiceLoader;
 import io.helidon.config.Config;
+import io.helidon.config.mp.MpConfig;
 import io.helidon.grpc.server.GrpcRouting;
 import io.helidon.grpc.server.GrpcServer;
 import io.helidon.grpc.server.GrpcServerConfiguration;
@@ -51,6 +52,7 @@ import io.helidon.microprofile.grpc.server.spi.GrpcMpExtension;
 import io.grpc.BindableService;
 import io.grpc.Channel;
 import io.grpc.inprocess.InProcessChannelBuilder;
+import org.eclipse.microprofile.config.ConfigProvider;
 
 /**
  * A CDI extension that will start the {@link GrpcServer gRPC server}.
@@ -73,7 +75,7 @@ public class GrpcServerCdiExtension
     private void startServer(@Observes @Initialized(ApplicationScoped.class) Object event, BeanManager beanManager) {
         GrpcRouting.Builder routingBuilder = discoverGrpcRouting(beanManager);
 
-        Config config = resolveConfig(beanManager);
+        Config config = MpConfig.toHelidonConfig(ConfigProvider.getConfig());
         GrpcServerConfiguration.Builder serverConfiguration = GrpcServerConfiguration.builder(config.get("grpc"));
         CompletableFuture<GrpcServer> startedFuture = new CompletableFuture<>();
         CompletableFuture<GrpcServer> shutdownFuture = new CompletableFuture<>();
@@ -126,26 +128,6 @@ public class GrpcServerCdiExtension
                           LOGGER.info(() -> "gRPC Server stopped in " + t + " milliseconds.");
                       }
                   });
-        }
-    }
-
-    /**
-     * Resolve the Helidon configuration to use.
-     * <p>
-     * The a bean of type {@link Config} is resolvable then that
-     * bean will be used for the configuration otherwise a new
-     * configuration will be created using {@link Config#create()}.
-     *
-     * @param beanManager  the {@link BeanManager} to use to resolve the config bean
-     *
-     * @return  the {@link Config} to use
-     */
-    private Config resolveConfig(BeanManager beanManager) {
-        Instance<Config> instance = beanManager.createInstance().select(Config.class);
-        if (instance.isResolvable()) {
-            return instance.get();
-        } else {
-            return Config.create();
         }
     }
 

--- a/microprofile/grpc/server/src/main/java/module-info.java
+++ b/microprofile/grpc/server/src/main/java/module-info.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2019, 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2019, 2021 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -25,6 +25,7 @@ module io.helidon.microprofile.grpc.server {
     requires transitive io.helidon.microprofile.grpc.core;
     requires io.helidon.common.serviceloader;
     requires io.helidon.microprofile.server;
+    requires io.helidon.config.mp;
 
     requires transitive io.grpc;
     requires grpc.protobuf.lite;

